insights/parsers/nginx_conf.py

```diff
@@ -0,0 +1,338 @@
+            for sub_item in item[1]:
+                subdict_all[sub_item[0]] = sub_item[1]
+            dict_all[item[0][0]] = subdict_all
+
+        if depth(item) == 5:
+            subdict_all = {}
+            for sub_item in item[1]:
+                if depth(sub_item) == 1:
+                    subdict_all[sub_item[0]] = sub_item[1]
+                if depth(sub_item) == 3:
+                    sub2dict_all = {}
+                    for sub2_item in sub_item[1]:
+                        sub2dict_all[sub2_item[0]] = sub2_item[1]
+                    if len(sub_item[0]) > 1:
+                        sub2dict_all["name"] = ' '.join(sub_item[0][1:])
+                    subdict_all[sub_item[0][0]] = _handle_key_value(subdict_all, sub_item[0][0], sub2dict_all)
+            dict_all[item[0][0]] = subdict_all
+
+        if depth(item) == 7:
+            subdict_all = {}
+            for sub_item in item[1]:
+                if depth(sub_item) == 1:
+                    subdict_all[sub_item[0]] = sub_item[1]
+                if depth(sub_item) == 3:
+                    sub2dict_all = {}
+                    for sub2_item in sub_item[1]:
+                        sub2dict_all[sub2_item[0]] = sub2_item[1]
+                    if len(sub_item[0]) > 1:
+                        sub2dict_all["name"] = ' '.join(sub_item[0][1:])
+                    subdict_all[sub_item[0][0]] = _handle_key_value(subdict_all, sub_item[0][0], sub2dict_all)
+                if depth(sub_item) == 5:
+                    sub3dict_all = {}
+                    for sub2_item in sub_item[1]:
+                        if depth(sub2_item) == 1:
+                            sub3dict_all[sub2_item[0]] = sub2_item[1]
+                        if depth(sub2_item) == 3:
+                            sub4dict_all = {}
+                            for sub3_item in sub2_item[1]:
+                                sub4dict_all[sub3_item[0]] = sub3_item[1]
+                            if len(sub2_item[0]) > 1:
+                                sub4dict_all["name"] = ' '.join(sub2_item[0][1:])
+                            sub3dict_all[sub2_item[0][0]] = _handle_key_value(sub3dict_all, sub2_item[0][0], sub4dict_all)
+                    subdict_all[sub_item[0][0]] = _handle_key_value(subdict_all, sub_item[0][0], sub3dict_all)
+            dict_all[item[0][0]] = subdict_all
+
+    return dict_all
+
+
+@parser('nginx_rh_conf')
+@parser('nginx_org_conf')
+class NginxConf(Parser, LegacyItemAccess):
+    """
+    Class for ``nginx.conf`` and ``conf.d`` configuration files.
+
+    Gerenally nginx.conf is writed as key-value format. It has a mail section and several sections,
+    http, mail, events, etc. They are unique, and subsection server and location in http section could
+    be duplicate, so the value of these subsections may be list.
+    """
+
+    def parse_content(self, content):
+        list1 = UnspacedList(create_parser().parseString("\n".join(get_active_lines(content))).asList())
```

Comment:
Can pyparsing take a list of lines?  It seems a bit expensive to join the content together again only for it to be parsed...  I'd also recommend splitting this up so it's a bit easier to read.